x-pack/plugins/enterprise_search/server/lib/indices/fetch_indices.ts

```diff
@@ -173,30 +80,40 @@ export const fetchIndices = async (
         privileges: { manage: false, read: false, ...indexPrivileges[name] },
       });
 
+      let expandedAliases = [] as ElasticsearchIndexWithPrivileges[];
       if (includeAliases) {
-        aliases.forEach((alias) => {
-          indicesAndAliases.push({
-            ...indexData,
-            alias: true,
-            count: indexCounts[alias] ?? 0,
-            name: alias,
-            privileges: { manage: false, read: false, ...indexPrivileges[name] },
-          });
-        });
+        expandedAliases = expandAliases(name, aliases, indexData, totalIndexData);
       }
-      return indicesAndAliases;
+      return !includeAliases ? indicesAndAliases : [...indicesAndAliases, ...expandedAliases];
```

Comment:
nitpick: we should be typing the variable, not typecasting the array (so: `const expandedAliases: ElasticsearchIndexWithPrivileges[] = [];`. 

Also, we can make this a little more immutable and a lot shorter by using triplets and array transforms, and then we probably don't need the explicit typing (this suggestion needs to include above up to line 74, but GitHub won't let me pull it all the way up 🤦):

```suggestion
      const indexEntry = {
        ...indexData,
        alias: false,
        count: indexCounts[name] ?? 0,
        name,
        privileges: { manage: false, read: false, ...indexPrivileges[name] },
      };
      return includeAliases ? [indexEntry, ...expandAliases(name, aliases, indexData, totalIndexData)] : [indexEntry];
```

Sidenote: it's almost always better to check for the positive case of a boolean rather than the negative case, it's easier to read and you're less likely to make a mistake.